infra/templates/azure-ase-app-service-single-region/docs/design.md

```diff
@@ -0,0 +1,161 @@
+| `acr_registry_server_username` | string | Username for acr registry **Note: required until ACR MSI supported via App Service** |
+| `acr_registry_server_password` | string | Password for acr registry **Note: required until ACR MSI supported via App Service** |
+
+### Firewall Implementation 
+Virtual Service Endpoints (VSE) will be used to restrict access to key services to a targeted VNET that is managed by the customer. An overview of this feature can be found [here](https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-service-endpoints-overview).
+
+**Note:** App Service VNET integration is managed for the customer through their existing ASE deployments. VSE is supported for this resource but it is not required for this project.
+
+| Azure Service | Supports VSE? | VSE Supported in Terraform? | Supported in Cobalt? |
+|---|---|---|---|
+| Key Vault | [Yes](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-overview-vnet-service-endpoints) | [Yes](https://www.terraform.io/docs/providers/azurerm/r/key_vault.html#network_acls) | [No](https://github.com/microsoft/cobalt/blob/master/infra/modules/providers/azure/keyvault/main.tf) - Missing Configuration |
+| Azure Storage | [Yes](https://docs.microsoft.com/en-us/azure/storage/common/storage-network-security?toc=%2fazure%2fvirtual-network%2ftoc.json#grant-access-from-a-virtual-network) | [Yes](https://www.terraform.io/docs/providers/azurerm/r/storage_account.html#virtual_network_subnet_ids) | [No](https://github.com/microsoft/cobalt/pull/183/files) - Missing configuration |
+| ACR | [Not for App Service](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-vnet) | N/A | No - **Feature Not Supported By Azure** |
+| Azure SQL | [Yes](https://docs.microsoft.com/en-us/azure/sql-database/sql-database-vnet-service-endpoint-rule-overview?toc=%2fazure%2fvirtual-network%2ftoc.json) | [Yes](https://www.terraform.io/docs/providers/azurerm/r/sql_virtual_network_rule.html#subnet_id) | No - Module Not Written |
+| Cosmos | [Yes](https://docs.microsoft.com/en-us/azure/cosmos-db/vnet-service-endpoint?toc=%2fazure%2fvirtual-network%2ftoc.json) | [Yes](https://www.terraform.io/docs/providers/azurerm/r/cosmosdb_account.html#virtual_network_rule) | [No](https://github.com/microsoft/cobalt/blob/master/infra/modules/providers/azure/cosmosdb/main.tf) - Missing Configuration |
+
+### Risks
+ - Currently ACR tasks are not supported for ACRs deployed to a virtual network. [Documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-vnet). This may be mitigated based on the customer's specific firewall configuration. Needs deep dive.
+ - Currently ACR and Virtual Service Endpoints are not compatible with App Services. [Documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-vnet)
+ - Currently ACR MSI access is not supported by App Services
+
+## Security
```

Comment:
May want to also reference the AuthN/AuthZ designs once we complete them.

Reply:
will add a note about this as an added security measure